customer/api.go

```diff
@@ -0,0 +1,178 @@
+package customer
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"sync"
+
+	"github.com/pkg/errors"
+)
+
+type API struct {
+	clientID   string
+	license    int32
+	HTTPClient *http.Client
+	ApiURL     string
+	token      string
+	tokenLock  sync.Mutex
+}
+
+func NewAPI(clientID string, license int32) *API {
+	return &API{
+		clientID:   clientID,
+		license:    license,
+		ApiURL:     "https://api.livechatinc.com/",
+		HTTPClient: http.DefaultClient,
```

Comment:
It'd be nice to be able to provide customized http client.
Also there should be some way to create api client for FRA.